---
title: 【Promise第32题】下面代码的输出是什么？
pubDatetime: 2022-01-09T23:15:24.000Z
author: caorushizi
tags:
  - JavaScript
postSlug: d2dabac22038970466717c35a69ee6c6
description: >-
  async function async1 () { console.log('async1 start'); await new
  Promise(resolve => { console.log('promise1') }) console.log('async1 success');
  return 'async1 end' } console.log('srcipt start') async
difficulty: 2.5
questionNumber: 1624
source: https://fe.ecool.fun/topic/52ba6f44-41dd-44cb-a2f4-2dc1589fb02b
---

```js
async function async1() {
  console.log("async1 start");
  await new Promise(resolve => {
    console.log("promise1");
  });
  console.log("async1 success");
  return "async1 end";
}
console.log("srcipt start");
async1().then(res => console.log(res));
console.log("srcipt end");
```

## 解析

在 async1 中 await 后面的 Promise 是没有返回值的，也就是它的状态始终是 pending 状态，因此相当于一直在 await，await，await 却始终没有响应...

所以在 await 之后的内容是不会执行的，也包括 async1 后面的 .then。

## 结果

```
'script start'
'async1 start'
'promise1'
'script end'
```
